include/linker/sections.h

```diff
@@ -62,15 +48,32 @@
 #define _DTCM_NOINIT_SECTION_NAME	.dtcm_noinit
 #endif
 
-#define IMX_BOOT_CONF	.boot_hdr.conf
-#define IMX_BOOT_DATA	.boot_hdr.data
-#define IMX_BOOT_IVT	.boot_hdr.ivt
-#define IMX_BOOT_DCD	.boot_hdr.dcd_data
+#define _IMX_BOOT_CONF_SECTION_NAME	.boot_hdr.conf
+#define _IMX_BOOT_DATA_SECTION_NAME	.boot_hdr.data
+#define _IMX_BOOT_IVT_SECTION_NAME	.boot_hdr.ivt
+#define _IMX_BOOT_DCD_SECTION_NAME	.boot_hdr.dcd_data
 
 #ifdef CONFIG_NOCACHE_MEMORY
 #define _NOCACHE_SECTION_NAME nocache
 #endif
 
+/* Short section references for use in ASM files */
+#if defined(_ASMLANGUAGE)
```

Comment:
I'm not very familiar with how these are used, but if there's a problem with collision, why not just delete all defines (TEXT, BSS, DATA, NOINIT) except TEXT_START and use the section names directly? Or use the longer defines above (e.g. _TEXT_SECTION_NAME) in the ASM files. AFAIU, if someone wants to use a different text section name, right now they have to change 2 defines instead of one.

Reply:
Technically, there is no reason why we cannot use `_TEXT_SECTION_NAME` and friends in the ASM files.

I do not know the full history of these symbols, but AFAICS, these short section names exist mainly for aesthetics.

For example, `SECTION_FUNC(TEXT, z_arm_pendsv)` looks cleaner than `SECTION_FUNC(_TEXT_SECTION_NAME, z_arm_pendsv)`.
https://github.com/zephyrproject-rtos/zephyr/blob/af3defd34e1c06e9fdc57da6c1b5c91a95c44023/arch/arm/core/aarch32/swap_helper.S#L52

Reply:
Ok, if we want to keep the short defines, it's probably best to define them like

    #define TEXT _TEXT_SECTION_NAME

Reply:
Only one problem with that approach: `DATA` != `_DATA_SECTION_NAME` -- doing so would introduce a functional change which I am trying to avoid in this PR.

https://github.com/zephyrproject-rtos/zephyr/blob/6829323036289d1a65d4e6c2f48a3259a8ebf81a/include/linker/sections.h#L21-L22

https://github.com/zephyrproject-rtos/zephyr/blob/6829323036289d1a65d4e6c2f48a3259a8ebf81a/include/linker/sections.h#L44

This goes all the way back to the ["first commit"](https://github.com/zephyrproject-rtos/zephyr/blame/master/include/linker/sections.h#L21-L22), and I cannot find any explanation for "Linker issue with XIP where the name "data" cannot be used."

@andrewboie @andyross Do you know if this is still relevant?

Reply:
I don't think anyone is still on the project who knows that history. It definitely pre-dates my involvement.

Reply:
@nashif do you know?

Reply:
This goes back to the first first commit, so it was in the code before it became zephyr..

```
Date:   Mon Nov 10 13:26:08 2014 -0500

    Initial set of VxMicro files
```

Reply:
I think we should leave these as is for now, since this PR is meant to be a simple clean-up.

As for the "data" vs "datas," I think we should address them when we do a proper refactoring/rework in the future.